### DISCUSSION TITLE
Proxying Plausible script - does it work on hosted websites?

### DISCUSSION THREAD

**Discussion by perfectly-preserved-pie** [plausible/analytics#2232] [Self-Hosted Support] [answered]
I've set up Plausible on self-hosted server and it's running successfully. I have a Jekyll blog hosted on GitHub Pages and a blog hosted on Tumblr. 

When I use `<script defer data-api="/api/event" data-domain="my.plausible.instance.com" src="/js/script.js"></script>` on either of them, nothing happens in the Plausible dashboard. I still get the spinning green dot. 

On my Tumblr blog, I see this error in the browser console:
`The resource from “https://my.tumblr.blog.comi/js/script.js” was blocked due to MIME type (“text/html”) mismatch (X-Content-Type-Options: nosniff)`

On my GitHub Pages blog, I see this:

`Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://my.plausible.instance.com/api/event. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing). Status code: 403.`

Is there anything I can do? Are these errors because of something the hosting provider is doing?

**Comment by littlebigco-dev** [comment, marked as answer]
The first issue refers to a mismatch of MIME type. Usually, script refers to JS. Your content  type option (nosniff) seems to prevent this and the JS is treated as HTML causing the issue. You either need to amend your content type option (Content-Type-Options header) or specify the JS as "type=javascript" in your code.

The second issue is because of the basic security setting to not execute code on foreign servers (CORS header). You need to tell the browser that it is OK to execute the script on your plausible installation. You need to set the Access-Control-Allow-Origin correctly.

Both settings need to be set by you, not the hosting provider. Depending on your host, they might need to set it up for you. If that is the case there should be a setting in your hosting panel to set those. Usually they are set in a htaccess file or directly via PHP for instance. In any case they need to be set serverside.
